docs/osf-trivia-list/deployment.md

```diff
@@ -12,20 +12,46 @@ Github](https://github.com/Dasharo/dasharo-issues).
 
 Following sections explain how to deal with most common `flashrom` problem.
 
-### How to install flashrom ?
+### How to install Dasharo flashrom fork ?
 
-* Install flashrom v1.1 or newer with your distribution's package manager if
-  you don't have it installed yet. If your distro doesn't provide flashrom or
-  provides an outdated one, you can build it yourself using
-  [this instruction](https://www.flashrom.org/Downloads).
-* Or compile recent version of flashrom:
+Following procedure is for advanced users who are familiar with source
+compilation and can resolve potential missing dependencies or other typical
+compilation problems. If you are not familiar with compilation from source we
+recommend use of Dasharo Tools Suite, which is minimal Linux environment to
+deploy, update, and maintain firmware on Dasharo-supported devices. Please
+follow [DTS documentation](../../dasharo-tools-suite/overview). Otherwise
+please follow compilation procedure.
 
-  ``` console
-  sudo apt install libpci-dev libftdi-dev libusb-1.0-0-dev
-  git clone https://github.com/flashrom/flashrom.git
-  cd flashrom
-  sudo make install
-  ```
+Currently, the latest flashrom release lacks support for Alder Lake
+S and Comet Lake U internal flashing. Because of this, we need to
+build flashrom from [Dashro
+fork](https://github.com/Dasharo/flashrom/tree/dasharo-release).
+The procedure was tested on [Ubuntu 22.04 desktop amd64
+ISO](https://releases.ubuntu.com/22.04/ubuntu-22.04.1-desktop-amd64.iso)
+burned on the USB stick. It may work on other Debian-based distributions.
+
+Boot the Ubuntu live image and select the `Try Ubuntu` option. Open a terminal
+and proceed with the commands below.
```

Comment:
Maybe we should resign from supporting compilation from source. Everyone should use our binary and if one would like to compile they are on they own?

Did you reported problem with Try Ubuntu anywhere? Maybe it is a bug?

Reply:
Well, this is definitely a solution, when we release binary for clients, we provide binaries, but you can build it yourself also.
I didn't build a flashrom on `Try Ubuntu` before. I used a prebuilt flashrom to avoid the need to install the whole system, although some packages needed to be installed anyway - it didn't work 100%. I don't know if it's a bug, but I think `Try Ubuntu` is intended for getting to know the system first, not for installing packages and performing more complicated operations on it.

Reply:
True, I agree with you. Key problem is that we have multiple version of flashrom compilation procedure and we cannot support all of them we can support maximum one if any, so if we agree that `Try Ubuntu` is not the way to go then we are left with question how in easy way advanced user can compile flashrom and under what OS, it can be DTS or other easy to setup environment (going through full OS installation is not easy IMHO). The barrier for getting through initial deployment procedure should be absolutely minimal.

@miczyg1 @macpijan what is you take on that?

Reply:
Ok, I'm in favor of removing it from documentation and getting rid of burden of maintaining that, @macpijan any objections?

Reply:
> I tried it on NovaCustom NS51MU and the system just crashes during installing packages.

That was a known Ubuntu bug of upgrading or installing packages with no `apt update`, IIRC. I do not have a reference on hand, but we can find it if needed.

> Ok, I'm in favor of removing it from documentation and getting rid of burden of maintaining that, @macpijan any objections?

Maybe have DTS option for "regular" users and custom-build flashrom for "advanced" users. But in there, we just inform which revision of flashrom we use. We do not bother with providing exact compliation steps, or prerequisites - we inform that one should refer to flashrom documentation?

Reply:
Flashrom compilation procedures should be always the same, so why to repeat original documentation?
What we have to provide is information what source should be compiled and hopefully we will unify our code enough to have one flashrom supporting all our cases and deliver binary in DTS. Upstream flashrom has its own issue, but what we need is good enough effect for Dasharo users, Dasahro developers should understand compilation process enough to deal with it.

Current instruction kind of suggest users to go through compilation which is user unfriendly.

Ergo, end user should not need to compile flashrom, if end user want that then following official documentation should be enough. If even in that case there are problems end user is on his/her own because we cannot support all possible compilation environments, which change over time (FreeBSD, OpenBSD, deb-based distro, rpm-based distro etc.).

Problem of bootstrapable and reproducible compilation environment is huge and not belong to this discussion IMHO.

Reply:
@macpijan https://github.com/Dasharo/docs/pull/513/commits/eec25a302107ec9d7aa7ed8ddc64d03e054c59c3

Reply:
@macpijan ping, only this conversation left before merging.

Reply:
Sorry I have missed that one. I'm pretty sure at least one customer will not be happy about that we remove Ubuntu compilation instructions. Let's see.